internal/osbuild/systemd_journald_stage.go

```diff
@@ -0,0 +1,104 @@
+/**
+Include the new journald config stage to configure journald to
+persist the journal.Instead of creating the
+directory `/var/log/journal`, we explicitly configure journald
+via the new stage.
+This is also what Fedora CoreOS like:
+https://github.com/coreos/fedora-coreos-config/blob/d709e4f5fc3970fb6b938dc7bcfdb8d602d2db94/overlay.d/05core/usr/lib/systemd/journald.conf.d/10-coreos-persistent.conf
+**/
+
+package osbuild
+
+import (
+	"fmt"
+	"regexp"
+)
+
+const configFilenameRegex = "^[a-zA-Z0-9_\\.-]{1,250}\\.conf$"
+
+type SystemdJournaldStageOptions struct {
+	Filename string                      `json:"filename"`
+	Config   SystemdJournaldConfigDropin `json:"config"`
+}
+
+func (SystemdJournaldStageOptions) isStageOptions() {}
+
+func (o SystemdJournaldStageOptions) validate() error {
+	filenameRegex := regexp.MustCompile(configFilenameRegex)
+	if !filenameRegex.MatchString(o.Filename) {
+		return fmt.Errorf("filename %q doesn't conform to schema (%s)", o.Filename, repoFilenameRegex)
+	}
+	if o.Config.Journal == (SystemdJournaldConfigJournalSection{}) {
+		return fmt.Errorf("the 'Journal' section is required")
+	}
+	if o.Config.Journal.Storage == "" && o.Config.Journal.Compress == "" && o.Config.Journal.SplitMode == "" && o.Config.Journal.MaxFileSec == "" && o.Config.Journal.MaxRetentionSec == "" && o.Config.Journal.SyncIntervalSec == 0 && o.Config.Journal.Audit == "" {
```

Comment:
Based on the condition in the previous line, will this ever be true?

Reply:
probably not , deleting it.